## Introduction to Data Science

#### University of Redlands - DATA 101
#### Prof: Joanna Bieri [joanna_bieri@redlands.edu](mailto:joanna_bieri@redlands.edu)
#### [Class Website: data101.joannabieri.com](https://joannabieri.com/data101.html)

---------------------------------------
# Homework Day 7
---------------------------------------

GOALS:

1. Answer all the questions from the lecture - we will use the scientists data and the grocery data.
2. Analyze data about College majors - your first EDA! (Think of this as a practice exam.)

----------------------------------------------------------

This homework has **9 Questions** and **A Practice Exam**

NOTE:
The practice exam is intended to give you an idea of what an exam in this class might feel like. Remember you will be expected to work on the exam **independently** and submit your work before we get together to work on the exam in class. **15\% of your Exam grade will come from what you can do independently** so start practicing now and get help if you need it!.

In [4]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.io as pio
pio.renderers.defaule = 'colab'

from itables import show

## Lecture Questions

First we will import the data about female scientists.

In [5]:
file1 = 'https://joannabieri.com/introdatascience/data/dates.csv'
DF_dates = pd.read_csv(file1)
file2 = 'https://joannabieri.com/introdatascience/data/professions.csv'
DF_professions = pd.read_csv(file2)
file3 = 'https://joannabieri.com/introdatascience/data/works.csv'
DF_works = pd.read_csv(file3)
show(DF_dates)
show(DF_professions)
show(DF_works)

**Q1** Are each of these data sets Tidy?

**Q2** Do they all contain the same number of observations?

**Q3** What are the five variables?

**Q4** What variable do they have in common - what can we join **on**

**Q1** They're all roughly tidy, with the third one being questionable due to listing many accolades in the same column.

**Q2** No, it varies from table to table.

**Q3** The 5 variables are 'name', 'birth_year', 'death_year', 'profession', and 'known_for'.

**Q4** They all have the 'name' variable.

## Combining Three Data Sets

You need to think about the order and whether or not you want to keep the maximum number of names or only keep names that are in all the data frames.

In this case we will try to keep the maximal amount of data - the maximum number of names.

1. Combine the professions and works data
2. Then add the dates data

In [11]:
# 1. Combine the professions and works data
DF_scientists = pd.merge(DF_professions,DF_works,on='name',how='left')
DF_scientists
# 2. Then add the dates data
DF_scientists = pd.merge(DF_scientists,DF_dates,on='name',how='left')
DF_scientists

name                          profession  \
0        Ada Lovelace                       Mathematician   
1         Marie Curie               Physicist and Chemist   
2        Janaki Ammal                            Botanist   
3     Chien-Shiung Wu                           Physicist   
4   Katherine Johnson                       Mathematician   
5   Rosalind Franklin                             Chemist   
6          Vera Rubin                          Astronomer   
7         Gladys West                       Mathematician   
8  Flossie Wong-Staal  Virologist and Molecular Biologist   
9     Jennifer Doudna                          Biochemist   

                                           known_for  birth_year  death_year  
0                           first computer algorithm         NaN         NaN  
1  theory of radioactivity,  discovery of element...         NaN         NaN  
2            hybrid species, biodiversity protection      1897.0      1984.0  
3  confim and refine theory of radioactive beta d...      1912.0      1997.0  
4  calculations of orbital mechanics critical to ...      1918.0      2020.0  
5                                                NaN      1920.0      1958.0  
6                           existence of dark matter      1928.0      2016.0  
7  mathematical modeling of the shape of the Eart...      1930.0         NaN  
8  first scientist to clone HIV and create a map ...      1947.0         NaN  
9  one of the primary developers of CRISPR, a gro...      1964.0         NaN

**Q5** What, if anything, would change if you switched to how='right'?

The people who appear in the data set is different. Also, Rosalind Franklin loses her job for some reason.

## Combining Two Data Sets - You Try

**Q6** Write code that would combine the professions data and the dates data, but drop any names that don't appear in both.

In [12]:
# Your code here
DF_scientists = pd.merge(DF_professions,DF_dates,on='name',how='inner')
DF_scientists

name                          profession  birth_year  \
0        Janaki Ammal                            Botanist        1897   
1     Chien-Shiung Wu                           Physicist        1912   
2   Katherine Johnson                       Mathematician        1918   
3   Rosalind Franklin                             Chemist        1920   
4          Vera Rubin                          Astronomer        1928   
5         Gladys West                       Mathematician        1930   
6  Flossie Wong-Staal  Virologist and Molecular Biologist        1947   
7     Jennifer Doudna                          Biochemist        1964   

   death_year  
0      1984.0  
1      1997.0  
2      2020.0  
3      1958.0  
4      2016.0  
5         NaN  
6         NaN  
7         NaN

## Combining Three Data Sets - You Try

**Q7** Try to combine the three data sets together (professions, works, and dates) but instead of doing what we did above, see if you can keep only the names that appeared in all three data sets.

In [15]:
# Your code here
DF_scientists = pd.merge(DF_dates, pd.merge(DF_professions, DF_dates, on='name', how='inner'), on='name', how='inner')
DF_scientists

name  birth_year_x  death_year_x  \
0        Janaki Ammal          1897        1984.0   
1     Chien-Shiung Wu          1912        1997.0   
2   Katherine Johnson          1918        2020.0   
3   Rosalind Franklin          1920        1958.0   
4          Vera Rubin          1928        2016.0   
5         Gladys West          1930           NaN   
6  Flossie Wong-Staal          1947           NaN   
7     Jennifer Doudna          1964           NaN   

                           profession  birth_year_y  death_year_y  
0                            Botanist          1897        1984.0  
1                           Physicist          1912        1997.0  
2                       Mathematician          1918        2020.0  
3                             Chemist          1920        1958.0  
4                          Astronomer          1928        2016.0  
5                       Mathematician          1930           NaN  
6  Virologist and Molecular Biologist          1947           NaN  
7                          Biochemist          1964           NaN

## Case study - Grocery Sales

Below you will load some data about grocery sales and see how joining or merging data can help us answer questions.

In [16]:
file1 = 'https://joannabieri.com/introdatascience/data/purchases.csv'
DF_purchases = pd.read_csv(file1)
file2 = 'https://joannabieri.com/introdatascience/data/prices.csv'
DF_prices = pd.read_csv(file2)
show(DF_purchases)
show(DF_prices)

## Calculate the total revenue

**Q8** Given the data above find the total revenue.

We will have to join the data frames so that we can see the price of what was sold. So first we look for a common column. In this case both data frames have **item** as a column.

Now, think about how to calculate revenue... we need to add up the total money we made, so we need to know the price of each thing sold.

**PAUSE - see if you can write some of the code for this before looking at the cells below**

In [76]:
DF_total_sold = DF_purchases.value_counts('item')

In [34]:
DF_total_sold = DF_total_sold.reset_index().rename(columns={"index": "value", 0: "count"})
DF_total_sold

item  count
0          milk      2
1        banana      1
2         bread      1
3  toilet paper      1

In [35]:
DF_price_map = pd.merge(DF_total_sold, DF_prices, on='item', how='inner')

In [36]:
DF_price_map['total_spent'] = DF_price_map['count'] * DF_price_map['price']

In [42]:
DF_price_map[['total_spent']].sum()

total_spent    5.75
dtype: float64

**See the lecture notes if you need heelp figuring this out!**

**Q9** Calculate the revenue per customer? Hint - group by the customer id and then apply the sum().

In [67]:
# Your code here
DF_bwoah = pd.merge(DF_purchases, DF_prices, on='item', how='left')
DF_bwoah.groupby(by=['customer_id']).sum()

item  price
customer_id                         
1             breadmilkbanana   1.95
2            milktoilet paper   3.80

# Practice Exam - EDA - College Majors

Your solutions should include code written to answer the questions AND words describing the results. For each problem:

1. Write code to get the answers. It is okay to copy and paste from our lecture notes or your homework assignments.
2. Describe what the code output is telling you.
3. Explain any interesting aspects of your results.

------------------------
-------------------------

The first step in the process of turning information into knowledge process is to summarize and describe the raw information - the data.
In this assignment we explore data on college majors and earnings, specifically the data begin the FiveThirtyEight story ["The Economic Guide To Picking A College Major"](https://fivethirtyeight.com/features/the-economic-guide-to-picking-a-college-major/).

These data originally come from the American Community Survey (ACS) 2010-2012 Public Use Microdata Series.
While this is outside the scope of this assignment, if you are curious about how raw data from the ACS were cleaned and prepared, see [the code](https://github.com/fivethirtyeight/data/blob/master/college-majors/college-majors-rscript.R) FiveThirtyEight authors used.

We should also note that there are many considerations that go into picking a major.
Earnings potential and employment prospects are two of them, and they are important, but they don't tell the whole story.
Keep this in mind as you analyze the data.

In [7]:
file_location = 'https://joannabieri.com/introdatascience/data/recent-grads.csv'
DF_raw = pd.read_csv(file_location)

In [56]:
DF_raw

Rank  Major_code                                      Major    Total  \
0       1        2419                      PETROLEUM ENGINEERING   2339.0   
1       2        2416             MINING AND MINERAL ENGINEERING    756.0   
2       3        2415                  METALLURGICAL ENGINEERING    856.0   
3       4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4       5        2405                       CHEMICAL ENGINEERING  32260.0   
..    ...         ...                                        ...      ...   
168   169        3609                                    ZOOLOGY   8409.0   
169   170        5201                     EDUCATIONAL PSYCHOLOGY   2854.0   
170   171        5202                        CLINICAL PSYCHOLOGY   2838.0   
171   172        5203                      COUNSELING PSYCHOLOGY   4626.0   
172   173        3501                            LIBRARY SCIENCE   1098.0   

         Men    Women            Major_category  ShareWomen  Sample_size  \
0     2057.0    282.0               Engineering    0.120564           36   
1      679.0     77.0               Engineering    0.101852            7   
2      725.0    131.0               Engineering    0.153037            3   
3     1123.0    135.0               Engineering    0.107313           16   
4    21239.0  11021.0               Engineering    0.341631          289   
..       ...      ...                       ...         ...          ...   
168   3050.0   5359.0    Biology & Life Science    0.637293           47   
169    522.0   2332.0  Psychology & Social Work    0.817099            7   
170    568.0   2270.0  Psychology & Social Work    0.799859           13   
171    931.0   3695.0  Psychology & Social Work    0.798746           21   
172    134.0    964.0                 Education    0.877960            2   

     Employed  ...  Part_time  Full_time_year_round  Unemployed  \
0        1976  ...        270                  1207          37   
1         640  ...        170                   388          85   
2         648  ...        133                   340          16   
3         758  ...        150                   692          40   
4       25694  ...       5180                 16697        1672   
..        ...  ...        ...                   ...         ...   
168      6259  ...       2190                  3602         304   
169      2125  ...        572                  1211         148   
170      2101  ...        648                  1293         368   
171      3777  ...        965                  2738         214   
172       742  ...        237                   410          87   

     Unemployment_rate  Median  P25th   P75th  College_jobs  Non_college_jobs  \
0             0.018381  110000  95000  125000          1534               364   
1             0.117241   75000  55000   90000           350               257   
2             0.024096   73000  50000  105000           456               176   
3             0.050125   70000  43000   80000           529               102   
4             0.061098   65000  50000   75000         18314              4440   
..                 ...     ...    ...     ...           ...               ...   
168           0.046320   26000  20000   39000          2771              2947   
169           0.065112   25000  24000   34000          1488               615   
170           0.149048   25000  25000   40000           986               870   
171           0.053621   23400  19200   26000          2403              1245   
172           0.104946   22000  20000   22000           288               338   

     Low_wage_jobs  
0              193  
1               50  
2                0  
3                0  
4              972  
..             ...  
168            743  
169             82  
170            622  
171            308  
172            192  

[173 rows x 21 columns]

### 1. Describe the data that you see here.

* How many variables?
* How man observations?
* Are the variables categorical or numerical?
* Are there any variables that you don't know what they mean? If so [CLICK HERE - to see the Git Repo for the data where the variables are defined](https://github.com/fivethirtyeight/data/tree/master/college-majors)


In [14]:
# DF_raw['abc'] = 111
DF_raw

Rank  Major_code                                      Major    Total  \
0       1        2419                      PETROLEUM ENGINEERING   2339.0   
1       2        2416             MINING AND MINERAL ENGINEERING    756.0   
2       3        2415                  METALLURGICAL ENGINEERING    856.0   
3       4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4       5        2405                       CHEMICAL ENGINEERING  32260.0   
..    ...         ...                                        ...      ...   
168   169        3609                                    ZOOLOGY   8409.0   
169   170        5201                     EDUCATIONAL PSYCHOLOGY   2854.0   
170   171        5202                        CLINICAL PSYCHOLOGY   2838.0   
171   172        5203                      COUNSELING PSYCHOLOGY   4626.0   
172   173        3501                            LIBRARY SCIENCE   1098.0   

         Men    Women            Major_category  ShareWomen  Sample_size  \
0     2057.0    282.0               Engineering    0.120564           36   
1      679.0     77.0               Engineering    0.101852            7   
2      725.0    131.0               Engineering    0.153037            3   
3     1123.0    135.0               Engineering    0.107313           16   
4    21239.0  11021.0               Engineering    0.341631          289   
..       ...      ...                       ...         ...          ...   
168   3050.0   5359.0    Biology & Life Science    0.637293           47   
169    522.0   2332.0  Psychology & Social Work    0.817099            7   
170    568.0   2270.0  Psychology & Social Work    0.799859           13   
171    931.0   3695.0  Psychology & Social Work    0.798746           21   
172    134.0    964.0                 Education    0.877960            2   

     Employed  ...  Unemployed  Unemployment_rate  Median  P25th   P75th  \
0        1976  ...          37           0.018381  110000  95000  125000   
1         640  ...          85           0.117241   75000  55000   90000   
2         648  ...          16           0.024096   73000  50000  105000   
3         758  ...          40           0.050125   70000  43000   80000   
4       25694  ...        1672           0.061098   65000  50000   75000   
..        ...  ...         ...                ...     ...    ...     ...   
168      6259  ...         304           0.046320   26000  20000   39000   
169      2125  ...         148           0.065112   25000  24000   34000   
170      2101  ...         368           0.149048   25000  25000   40000   
171      3777  ...         214           0.053621   23400  19200   26000   
172       742  ...          87           0.104946   22000  20000   22000   

     College_jobs  Non_college_jobs  Low_wage_jobs  abc  xyz  
0            1534               364            193  111  222  
1             350               257             50  111  222  
2             456               176              0  111  222  
3             529               102              0  111  222  
4           18314              4440            972  111  222  
..            ...               ...            ...  ...  ...  
168          2771              2947            743  111  222  
169          1488               615             82  111  222  
170           986               870            622  111  222  
171          2403              1245            308  111  222  
172           288               338            192  111  222  

[173 rows x 23 columns]

In [13]:
DF_copy = DF_raw
DF_copy['xyz'] = 222
DF_copy

Rank  Major_code                                      Major    Total  \
0       1        2419                      PETROLEUM ENGINEERING   2339.0   
1       2        2416             MINING AND MINERAL ENGINEERING    756.0   
2       3        2415                  METALLURGICAL ENGINEERING    856.0   
3       4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4       5        2405                       CHEMICAL ENGINEERING  32260.0   
..    ...         ...                                        ...      ...   
168   169        3609                                    ZOOLOGY   8409.0   
169   170        5201                     EDUCATIONAL PSYCHOLOGY   2854.0   
170   171        5202                        CLINICAL PSYCHOLOGY   2838.0   
171   172        5203                      COUNSELING PSYCHOLOGY   4626.0   
172   173        3501                            LIBRARY SCIENCE   1098.0   

         Men    Women            Major_category  ShareWomen  Sample_size  \
0     2057.0    282.0               Engineering    0.120564           36   
1      679.0     77.0               Engineering    0.101852            7   
2      725.0    131.0               Engineering    0.153037            3   
3     1123.0    135.0               Engineering    0.107313           16   
4    21239.0  11021.0               Engineering    0.341631          289   
..       ...      ...                       ...         ...          ...   
168   3050.0   5359.0    Biology & Life Science    0.637293           47   
169    522.0   2332.0  Psychology & Social Work    0.817099            7   
170    568.0   2270.0  Psychology & Social Work    0.799859           13   
171    931.0   3695.0  Psychology & Social Work    0.798746           21   
172    134.0    964.0                 Education    0.877960            2   

     Employed  ...  Unemployed  Unemployment_rate  Median  P25th   P75th  \
0        1976  ...          37           0.018381  110000  95000  125000   
1         640  ...          85           0.117241   75000  55000   90000   
2         648  ...          16           0.024096   73000  50000  105000   
3         758  ...          40           0.050125   70000  43000   80000   
4       25694  ...        1672           0.061098   65000  50000   75000   
..        ...  ...         ...                ...     ...    ...     ...   
168      6259  ...         304           0.046320   26000  20000   39000   
169      2125  ...         148           0.065112   25000  24000   34000   
170      2101  ...         368           0.149048   25000  25000   40000   
171      3777  ...         214           0.053621   23400  19200   26000   
172       742  ...          87           0.104946   22000  20000   22000   

     College_jobs  Non_college_jobs  Low_wage_jobs  abc  xyz  
0            1534               364            193  111  222  
1             350               257             50  111  222  
2             456               176              0  111  222  
3             529               102              0  111  222  
4           18314              4440            972  111  222  
..            ...               ...            ...  ...  ...  
168          2771              2947            743  111  222  
169          1488               615             82  111  222  
170           986               870            622  111  222  
171          2403              1245            308  111  222  
172           288               338            192  111  222  

[173 rows x 23 columns]

In [6]:
DF_copy = DF_raw.copy()

NameError: name 'DF_raw' is not defined

There are 21 variables and 173 observations.
19 variables are numerical while 2 are categorical.
Major_code was confusing to me at first but it makes sense after reading the documentation.

### 2. Answer some questions about the data

* Which major has the lowest unemployment rate? Show a data frame that answers this question, but only show columns that are useful to the answer.
* Which major has the highest percentage of women? Show a data frame that answers this question, but only show columns that are useful to the answer.
* How do the distributions of median income compare across major categories?
    * First get a value count on what major categories are in the data. What categories are most or least represented?
    * Do some summary statistics using the .group_by() command to see the statistics for each major category.
    * Make a histogram that explores median income for each of the major categories (facet_col). How did you choose your number of bins?
    * Why do we often choose the median, rather than the mean, to describe the typical income of a group of people? Look this up online if you don't know the answer!
    * Answer the question: Which major category has the highest typical (you'll need to decide what this means) median income?

In [64]:
DF_unemployment = DF_raw[['Major', 'Employed', 'Unemployed', 'Unemployment_rate']].sort_values(by='Unemployment_rate')
DF_unemployment

Major  Employed  Unemployed  \
52             MATHEMATICS AND COMPUTER SCIENCE       559           0   
112                                SOIL SCIENCE       613           0   
83                                       BOTANY      1010           0   
73                        MILITARY TECHNOLOGIES         0           0   
120  EDUCATIONAL ADMINISTRATION AND SUPERVISION       703           0   
..                                          ...       ...         ...   
29                                PUBLIC POLICY      4547         670   
170                         CLINICAL PSYCHOLOGY      2101         368   
84   COMPUTER NETWORKING AND TELECOMMUNICATIONS      6144        1100   
89                        PUBLIC ADMINISTRATION      4158         789   
5                           NUCLEAR ENGINEERING      1857         400   

     Unemployment_rate  
52            0.000000  
112           0.000000  
83            0.000000  
73            0.000000  
120           0.000000  
..                 ...  
29            0.128426  
170           0.149048  
84            0.151850  
89            0.159491  
5             0.177226  

[173 rows x 4 columns]

In [65]:
DF_women = DF_raw[['Major', 'Total', 'Men', 'Women', 'ShareWomen']].sort_values(by='ShareWomen')
DF_women

Major     Total      Men  \
73                           MILITARY TECHNOLOGIES     124.0    124.0   
66     MECHANICAL ENGINEERING RELATED TECHNOLOGIES    4790.0   4419.0   
26                           CONSTRUCTION SERVICES   18498.0  16820.0   
1                   MINING AND MINERAL ENGINEERING     756.0    679.0   
3        NAVAL ARCHITECTURE AND MARINE ENGINEERING    1258.0   1123.0   
..                                             ...       ...      ...   
138                           ELEMENTARY EDUCATION  170862.0  13029.0   
51                      MEDICAL ASSISTING SERVICES   11123.0    803.0   
163  COMMUNICATION DISORDERS SCIENCES AND SERVICES   38279.0   1225.0   
164                      EARLY CHILDHOOD EDUCATION   37589.0   1167.0   
21                                    FOOD SCIENCE       NaN      NaN   

        Women  ShareWomen  
73        0.0    0.000000  
66      371.0    0.077453  
26     1678.0    0.090713  
1        77.0    0.101852  
3       135.0    0.107313  
..        ...         ...  
138  157833.0    0.923745  
51    10320.0    0.927807  
163   37054.0    0.967998  
164   36422.0    0.968954  
21        NaN         NaN  

[173 rows x 5 columns]

In [89]:
DF_category = DF_raw.value_counts('Major_category')
DF_category = DF_category.reset_index().rename(columns={"index": "value", 0: "count"})
DF_category

Major_category  count
0                           Engineering     29
1                             Education     16
2             Humanities & Liberal Arts     15
3                Biology & Life Science     14
4                              Business     13
5                                Health     12
6               Computers & Mathematics     11
7       Agriculture & Natural Resources     10
8                     Physical Sciences     10
9              Psychology & Social Work      9
10                       Social Science      9
11                                 Arts      8
12  Industrial Arts & Consumer Services      7
13                  Law & Public Policy      5
14          Communications & Journalism      4
15                    Interdisciplinary      1

In [106]:
my_columns = list(DF_raw.keys())
my_columns.remove('Major')

DF_raw.groupby(by=['Major_category'])[my_columns].describe()

Rank                                \
                                    count        mean        std    min   
Major_category                                                            
Agriculture & Natural Resources      10.0  101.500000  45.132521   22.0   
Arts                                  8.0  131.125000  45.091772   33.0   
Biology & Life Science               14.0   95.357143  33.726518   49.0   
Business                             13.0   55.846154  31.929209    7.0   
Communications & Journalism           4.0  104.000000  14.809907   94.0   
Computers & Mathematics              11.0   57.000000  24.195041   21.0   
Education                            16.0  130.375000  27.436897   56.0   
Engineering                          29.0   22.620690  18.640229    1.0   
Health                               12.0   96.500000  40.435133   35.0   
Humanities & Liberal Arts            15.0  135.066667  27.698547   70.0   
Industrial Arts & Consumer Services   7.0  105.142857  48.933385   27.0   
Interdisciplinary                     1.0  110.000000        NaN  110.0   
Law & Public Policy                   5.0   64.600000  36.411537   20.0   
Physical Sciences                    10.0   67.600000  31.549608    8.0   
Psychology & Social Work              9.0  143.000000  36.390246   71.0   
Social Science                        9.0   91.666667  36.231892   37.0   

                                                                  Major_code  \
                                        25%    50%     75%    max      count   
Major_category                                                                 
Agriculture & Natural Resources       66.75  110.0  136.25  162.0       10.0   
Arts                                 130.50  148.5  155.50  167.0        8.0   
Biology & Life Science                71.25   88.5  119.50  169.0       14.0   
Business                              36.00   58.0   77.00  128.0       13.0   
Communications & Journalism           96.25   98.0  105.75  126.0        4.0   
Computers & Mathematics               44.50   48.0   68.00  106.0       11.0   
Education                            119.50  132.5  145.75  173.0       16.0   
Engineering                           10.00   17.0   31.00   67.0       29.0   
Health                                58.75  104.5  123.25  164.0       12.0   
Humanities & Liberal Arts            116.50  138.0  158.50  168.0       15.0   
Industrial Arts & Consumer Services   75.00  107.0  145.50  161.0        7.0   
Interdisciplinary                    110.00  110.0  110.00  110.0        1.0   
Law & Public Policy                   30.00   88.0   90.00   95.0        5.0   
Physical Sciences                     45.50   74.0   89.75  111.0       10.0   
Psychology & Social Work             146.00  156.0  170.00  172.0        9.0   
Social Science                        69.00   80.0  125.00  143.0        9.0   

                                                  ... Non_college_jobs  \
                                            mean  ...              75%   
Major_category                                    ...                    
Agriculture & Natural Resources      1152.500000  ...          5535.25   
Arts                                 6015.125000  ...         29770.75   
Biology & Life Science               3475.857143  ...          6210.25   
Business                             6213.000000  ...         48447.00   
Communications & Journalism          1902.500000  ...         44495.00   
Computers & Mathematics              2702.727273  ...         10288.50   
Education                            2387.687500  ...          7843.75   
Engineering                          2489.896552  ...          3896.00   
Health                               5937.916667  ...         10275.25   
Humanities & Liberal Arts            4048.600000  ...         19253.00   
Industrial Arts & Consumer Services  4315.285714  ...         13758.50   
Interdisciplinary                    4000.000000  ... 

**(Click Here to Explain your Results)**

### 3. More complicated questions

* Are all STEM fields the same in terms job opportunities?
    * Assume the following are STEM major categories
    
  ``` ["Biology & Life Science", "Computers & Mathematics", "Engineering","Physical Sciences"]```
  
    * Create a DataFrame (using a mask) so you only look at STEM majors.
    * Which STEM majors have median salaries equal to or less than the median for all majors' median earnings?
* What types of majors do women tend to major in?
    * Create a scatterplot of median income vs. proportion of women in that major, coloured by whether the major is in a STEM field or not.

In [3]:
mask = DF_raw['Major_category'] = [['Biology & Life Science', 'Computers & Mathematics', 'Engineering', 'Physical Sciences']]
DF_raw[mask]

NameError: name 'DF_raw' is not defined

In [22]:
#

In [23]:
#

**(Click Here to Explain your Results)**

### 4. Ask a question of interest to you.

In [24]:
#

**(Click Here to Explain your Results)**